### PR TITLE
Add SEO preview card for Categories

### PR DESCRIPTION
**Pull request by sarjon**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | It adds SEO preview card in Categories form. See screenshot below.
| Type?         | improvement
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | https://github.com/PrestaShop/PrestaShop/issues/13259
| How to test?  | See below.

### How to test?
:warning: Rebuild assets.

1. Every time Meta title, Meta description or Friendly URL is changed the card should update.
2. When Meta title or Meta description is empty it falls back to Name or Description.
3. It should work it multiple languages as well.

### Screenshot

category_seo_preview

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/13132)

### CODE REVIEW

**Comment by sarjon** [comment]
Call for help:
1. @LouiseBonnard could you check wording? I've copied SEO preview card text from Product page, so it's not relevant for Categories and needs to be updated.
2. @TristanLDD could you check UI/UX?

**Comment by LouiseBonnard** [comment]
@sarjon, nice! I would write it the same way, see below and let me know:

_Improve your ranking and how your category page will appear in search engines results._

_Here is a preview of your search engine result, play with it!_

**Comment by TristanLDD** [comment]
Hello @sarjon, the only UI fix I see on this page is the Choose files button, we should use the one from the UI Kit (https://cdn.rawgit.com/PrestaShop/prestashop-ui-kit/master/index.html)
Capture d’écran 2019-04-01 à 16 18 53

**Comment by sarjon** [comment]
This PR is ready for review/testing. Although I'm not sure if the behavior is correct, I need confirmation regarding that. ^^

File: admin-dev/themes/new-theme/js/components/seo-preview-card.js [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+export default class SeoPreviewCard {
+  constructor(
+    seoPreviewCardSelector,
+    titleInputSelector,
+    urlInputSelector,
+    descriptionInputSelector,
+    fallbackTitleInputSelector,
+    fallbackDescriptionInputSelector,
+  ) {
+    console.log(titleInputSelector);
```
Comment (PierreRambaud) [inline admin-dev/themes/new-theme/js/components/seo-preview-card.js]: remove

File: admin-dev/themes/new-theme/js/components/seo-preview-card.js [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+   * @param {String} langId
+   *
+   * @private
+   */
+  _refreshCard(langId) {
+    if (this.request) {
+      this.request.abort();
+    }
+
+    this.request = $.ajax(this.$card.data('seo-preview-url-generator'), {
```
Comment (PierreRambaud) [inline admin-dev/themes/new-theme/js/components/seo-preview-card.js]: Why not using $.getJSON instead?

File: admin-dev/themes/new-theme/js/components/seo-preview-card.js [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+  }
+
+  /**
+   * Gets description in given language
+   *
+   * @param {String} langId
+   *
+   * @private
+   */
+  _getDescription(langId) {
```
Comment (PierreRambaud) [inline admin-dev/themes/new-theme/js/components/seo-preview-card.js:131]: we have already the serp component in vuejs to do SEO preview :thinking:

**Comment by PierreRambaud** [comment]
Hum, I think we should reuse what have already been done by @Quetzacoalt91 for the product page.

**Comment by Quetzacoalt91** [comment]
If your version is better, mine can be removed. But as @PierreRambaud said, code must be reused for the same feature.

**Comment by sarjon** [comment]
> If your version is better, mine can be removed. But as @PierreRambaud said, code must be reused for the same feature.

I've tried to reuse it, but it does not work with multilanguage & I'm not very familiar with Vue.js. And there was some other issue that I can't remember now. :thinking:

File: admin-dev/themes/new-theme/js/components/seo-preview-card.js (hunk 2 above)
Comment (sarjon) [reply admin-dev/themes/new-theme/js/components/seo-preview-card.js]: No reason, I can use `getJSON` if it's any better. :)

**Comment by PierreRambaud** [comment]
It's maybe the time to improve the old one :)

**Comment by sarjon** [comment]
> It's maybe the time to improve the old one :)

I agree, but the problem is there are not a lot of developers who knows Vue very well. I could do that, but I need more experience with Vue on some side projects first. ^^ So I don't want to introduce spaghetti Vue. :smiling_imp:

File: admin-dev/themes/new-theme/js/components/seo-preview-card.js (hunk 1 above)
Comment (sarjon) [reply admin-dev/themes/new-theme/js/components/seo-preview-card.js]: done.

File: admin-dev/themes/new-theme/js/components/seo-preview-card.js (hunk 2 above)
Comment (sarjon) [reply admin-dev/themes/new-theme/js/components/seo-preview-card.js]: done.

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Categories/Blocks/form.html.twig [hunk 4]
```diff
@@ -123,6 +123,28 @@
 
       {{ renderhook('displayBackOfficeCategory') }}
 
+      <div class="form-group row">
+        <label class="form-control-label"></label>
+        <div class="col-sm">
+          <h2>{{ 'Search Engine Optimization'|trans({}, 'Admin.Catalog.Feature') }}</h2>
+          <p class="mb-2">
+            {{ ' Improve your ranking and how your category page will appear in search engines results.'|trans({}, 'Admin.Catalog.Feature') }}
```
Comment (LouiseBonnard) [inline src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Categories/Blocks/form.html.twig:131]: Watch the space at the beginning. ;-)

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Categories/Blocks/form.html.twig [hunk 5]
```diff
@@ -123,6 +123,28 @@
...
+          </p>
+          <p>{{ "Here is a preview of your search engine result, play with it!"|trans({}, 'Admin.Catalog.Feature') }}</p>
```
Comment (LouiseBonnard) [inline src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Categories/Blocks/form.html.twig:133]: Is it the right form of quotation marks here?

Comment (sarjon) [reply src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Categories/Blocks/form.html.twig:133]: Good catch, it's not. :)

**Comment by matks** [comment]
@PierreRambaud will you be able to help us on this one ? 😉

**Comment by PierreRambaud** [comment]
@matks Of course, is it mandatory for 1.7.6?

**Comment by sarjon** [comment]
@matks what do we do with this PR?

**Comment by matks** [comment]
> @matks what do we do with this PR?

It seems @PierreRambaud and @eternoendless want to use the VueJS component after improving it so let's wait for them 😉 .

**Comment by sarjon** [comment]
Closing in favor of https://github.com/PrestaShop/PrestaShop/pull/13714

**Pull request closed without merging**